### ISSUE TITLE
Can wanda be used to ConvNet for CV tasks?

### ISSUE THREAD

**Issue by frankinwi** [locuslab/wanda#32]
Hello. Thanks for sharing your great work.

Appendix A states "we only prune the linear layers for ConvNeXt." I understand the motivation of wanda builds upon the outliers observed in LLM, where there are no Conv layers. However, for ConvNet (e.g., ResNet) in CV tasks, the effect of outliers might not be as significant as that in LLM. I wonder whether wanda can achieve a good performance for ConvNet without retraining. 

Thank you.

**Comment by Eric-mingjie** [comment]
Hi, thanks for the question. For ConvNeXt, we only prune the 1x1 convolution, which are implemented as linear FC layers. This is because they take up about most of the parameters in the ConvNeXt. We didn't try pruning convolution layers with larger kernels, as used by standard ResNets, but our method should be applicable. I agree that activations would play a more important role for pruning LLMs, where outliers are more prominent.

**Comment by frankinwi** [comment]
@Eric-mingjie Thanks for your reply. I have reproduced wanda on some LLM, and it is indeed "simple and effective" as the paper title emphasizes.

The pruning metric `S=|W|x||X||` should be applicable to generate the pruning mask for convolution layers, just like magnitude-based pruning. But I guess it might require retraining or finetuning to recover the accuracy for CV tasks.

At last, I appreciate this work for its simplicity and effectiveness.

----------------------------------------

## 2024.1.9 Reproduced results for DeiT on ImageNet-1K.

| Model  | Pretrained | Wanda (Sparisty 50%) |
|----------------|----------------|---------------|
|   deit_tiny_patch16_224   |   72.2    |       55.456   | 
|   deit_small_patch16_224  |  79.8   |      68.634    | 
|   deit_base_patch16_224   |  81.8   |      78.360    |
